### PR TITLE
[MC] Optimize loops in MC.

### PR DESCRIPTION
**Pull request by chestnykh**
https://llvm.org/docs/CodingStandards.html tells us that we should avoid evaluating `.end()` each time if possible.

### CODE REVIEW

**Review by s-barannikov — APPROVED** [review]
(nit) We don't usually end the subject line of a commit message with a period.

File: llvm/lib/MC/XCOFFObjectWriter.cpp [hunk 1]
```diff
@@ -1385,23 +1385,25 @@ void XCOFFObjectWriter::addExceptionEntry(
 unsigned XCOFFObjectWriter::getExceptionSectionSize() {
   unsigned EntryNum = 0;
 
-  for (auto it = ExceptionSection.ExceptionTable.begin();
-       it != ExceptionSection.ExceptionTable.end(); ++it)
+  for (auto I = ExceptionSection.ExceptionTable.begin(),
```
Comment (MaskRay) [inline llvm/lib/MC/XCOFFObjectWriter.cpp:1388]: this can use `for each`

**Comment by chestnykh** [comment]
> (nit) We don't usually end the subject line of a commit message with a period.

Got it

File: llvm/lib/MC/XCOFFObjectWriter.cpp (hunk 1 above)
Comment (chestnykh) [reply llvm/lib/MC/XCOFFObjectWriter.cpp]: Rewritten to `for_each`

**Review by nikic — COMMENTED** [review]
Can't some/all of these be range based for loops instead?

**Comment by chestnykh** [comment]
> Can't some/all of these be range based for loops instead?

Is range based loop preferable?

> Can't some/all of these be range based for loops instead?

I left comments where the loop can be transformed into range-based. Other loops use iterator in their body so i think we can leave them without changes.

**Comment by nikic** [comment]
> > Can't some/all of these be range based for loops instead?
> 
> Is range based loop preferable?

Yes :) From the same guide: https://llvm.org/docs/CodingStandards.html#use-range-based-for-loops-wherever-possible

File: llvm/lib/MC/XCOFFObjectWriter.cpp [hunk 2]
```diff
@@ -1385,23 +1385,23 @@ void XCOFFObjectWriter::addExceptionEntry(
 unsigned XCOFFObjectWriter::getExceptionSectionSize() {
   unsigned EntryNum = 0;
 
-  for (auto it = ExceptionSection.ExceptionTable.begin();
```
Comment (s-barannikov) [inline llvm/lib/MC/XCOFFObjectWriter.cpp:1388]: Can this be range-based too?
Please restore the comment.

**Review by s-barannikov — APPROVED** [review]
LGTM
A note for the future: amending commits and force pushing are discouraged, see
https://llvm.org/docs/GitHub.html#updating-pull-requests

**Comment by chestnykh** [comment]
> LGTM A note for the future: amending commits and force pushing are discouraged, see https://llvm.org/docs/GitHub.html#updating-pull-requests

FYI i don't have commit access so please commit my PR when it is ready

**Pull request merged**